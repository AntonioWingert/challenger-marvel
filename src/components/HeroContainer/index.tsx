import { MarvelApiResponse } from '../../@types/Api';
import Hero from '../Hero';
import * as S from './styles';

export default function HeroContainer(result: MarvelApiResponse) {
  return (
    <S.Container>
      {result.data.results.map((hero) => (
        <Hero
          key={hero.id}
          id={hero.id}
          name={hero.name}
          thumbnail={hero.thumbnail.path}
        />
      ))}
    </S.Container>
  );
}