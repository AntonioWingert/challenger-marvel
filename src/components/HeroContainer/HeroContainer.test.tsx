import { screen } from '@testing-library/react';
import render from '../../utils/renderWithRouter';
import HeroContainer from '.';
import mockMarvelAPI from '../../mocks/HeroData.mock';

describe('<Hero />', () => {
  it('should render correctly with default props', () => {
    const { container } = render(
      <HeroContainer data={mockMarvelAPI.data}/>
    );

    expect(screen.getByText(/iron man/i)).toBeVisible();
    expect(screen.getByAltText('Hero image Iron Man')).toBeVisible();
    expect(screen.getByAltText('Hero image Iron Man')).toHaveAttribute('src', 'http://example.com/ironman/portrait_medium.jpg');
    expect(container.firstChild).toMatchSnapshot();
  });
});