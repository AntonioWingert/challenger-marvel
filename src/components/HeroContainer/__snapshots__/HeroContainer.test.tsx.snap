// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Hero /> > should render correctly with default props 1`] = `
.c1 {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  gap: 0.2rem;
  width: 150px;
  height: 270px;
  background-color: #ccc;
  border-radius: 0.5rem;
}

.c1 h2 {
  font-size: 1rem;
  text-align: center;
}

.c1 img {
  width: 100px;
  height: 150px;
}

.c1 a {
  display: flex;
  align-items: center;
  justify-content: center;
  text-decoration: none;
  color: #F5F5F5;
  width: 100%;
  height: 25px;
  background-color: #4A1414;
}

.c0 {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  flex-wrap: wrap;
  gap: 1rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h2>
      Iron Man
    </h2>
    <img
      alt="Hero image Iron Man"
      src="http://example.com/ironman/portrait_medium.jpg"
    />
    <a
      href="/description/1"
    >
      Ver Mais
    </a>
  </div>
  <div
    class="c1"
  >
    <h2>
      Spider Man
    </h2>
    <img
      alt="Hero image Spider Man"
      src="http://example.com/spiderman/portrait_medium.jpg"
    />
    <a
      href="/description/2"
    >
      Ver Mais
    </a>
  </div>
</div>
`;
